grpc/src/builder.rs

```diff
@@ -0,0 +1,179 @@
+use std::error::Error as StdError;
+
+use k256::ecdsa::VerifyingKey;
+use signature::Keypair;
+use tonic::body::Body as TonicBody;
+use tonic::codegen::Service;
+
+use crate::boxed::{boxed, AbstractBody, BoxedTransport, ConditionalSend};
+use crate::client::SignerConfig;
+use crate::signer::DispatchedDocSigner;
+use crate::{DocSigner, GrpcClient, GrpcClientBuilderError};
+
+use imp::build_transport;
+
+enum TransportSetup {
+    Configuration { url: String, tls: bool },
+    BoxedTransport(BoxedTransport),
+}
+
+/// Builder for [`GrpcClient`]
+pub struct GrpcClientBuilder {
+    transport: TransportSetup,
+    signer_bits: Option<SignerConfig>,
+}
+
+impl GrpcClientBuilder {
+    /// Create a new client connected to the given `url` using [`Channel`] transport
+    ///
+    /// [`Channel`]: tonic::transport::Channel
+    pub fn with_url(url: impl Into<String>) -> Self {
+        GrpcClientBuilder {
+            transport: TransportSetup::Configuration {
+                url: url.into(),
+                tls: false,
+            },
+            signer_bits: None,
+        }
+    }
+
+    /// Create a gRPC client builder using provided prepared transport
+    pub fn with_transport<B, T>(transport: T) -> Self
+    where
+        //B: Body<Data = Bytes> + Send + Sync + Unpin + 'static,
+        B: http_body::Body + AbstractBody + Send + Unpin + 'static,
```

Comment:
I just tested locally and the line without the `AbstractBody` compliles. Is there any reason you left it there? I prefer all the boxed mechanism to be hidden from the user.